xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-macro/src/main/resources/templates/liveData/macro.vm

```diff
@@ -0,0 +1,32 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+## Since: 16.0.0RC1
+## Provide the macro to display a Live Data using Velocity.
+## $parameters the same parameters are the live data macro, but as an object instead of a set of macro parameters
+## $advancedParameters the advanced parameters usually located in the macro body, but as an object
+## $syntax the syntax to use for the rendering of the result
+#macro (liveData $parameters $advancedParameters $syntax)
```

Comment:
I wonder if it wouldn't be cleaner to provide this simple helper macro directly in the script service.

Reply:
Actually, I just noticed that you're serializing an object to JSON here and then later this is deserialized again. Couldn't we avoid that JSON roundtrip here?